### DISCUSSION TITLE
Polyfilling String.prototype.replaceAll

### DISCUSSION THREAD

**Discussion by kripod** [vercel/next.js#15715] [Help] [answered]
As far as I know, Next.js ships with its own set of polyfills instead of relying upon`corejs@3` through `@babel/preset-env`. What is the idiomatic way for making `String.prototype.replaceAll` available to IE11 users?

**Comment by jamesmosier** [comment, marked as answer]
Next recommends adding additional polyfills in your `_app.js` or in the component itself. [Custom Polyfill docs](https://nextjs.org/docs/basic-features/supported-browsers-features#custom-polyfills)

**Comment by kripod** [comment]
Thank you! Is there a chance that Next.js will implement `replaceAll` anytime soon?

**Reply by timneutkens** [reply]
At this point it's quite unlikely given that browser support is not there: https://caniuse.com/#search=replaceAll

We'd have to ship `replaceAll` to all browsers instead of just legacy ones given that edge does not support it.

**Comment by kripod** [comment]
Thanks for the answers! Where may I find a list of which functionality is polyfilled? I’m thinking about configuring [eslint-plugin-compat](https://github.com/amilajack/eslint-plugin-compat) based upon it.

**Reply by timneutkens** [reply]
The full list is available here: https://github.com/vercel/next.js/blob/canary/packages/next-polyfill-nomodule/src/index.js

**Reply by timneutkens** [reply]
These are only loaded in legacy browsers (the ones that do not support esmodules)

**Reply by kripod** [reply]
Wonderful, thank you! 🙏
